core/commands/generate_command.rb

```diff
@@ -212,37 +214,63 @@ def make_product_config(product_name, product, box, repo)
     { "#{attribute_name}": config }
   end
+  # rubocop:enable Metrics/CyclomaticComplexity
 
   # Make the list of the role parameters in the JSON-format.
   #
   # @param name [String] internal name of the machine specified in the template
   # @param product_config [Hash] list of the product parameters
   # @param recipe_name [String] name of the recipe
+  # @param subscription_manager_params [Hash] subscription manager params
+  # in format { need_configure, recipe_name, credentials }
   # @return [String] pretty formatted role description in JSON format.
-  def make_role_json(name, product_config, recipe_name)
-    role = {
-      name: name,
-      default_attributes: {},
-      override_attributes: product_config,
-      json_class: 'Chef::Role',
-      description: '',
-      chef_type: 'role',
-      run_list: ['recipe[mdbci_provision_mark::remove_mark]',
-                 "recipe[#{recipe_name}]",
-                 'recipe[mdbci_provision_mark::default]']
-    }
+  def make_role_json(name, product_config, recipe_name, subscription_manager_params)
+    run_list = ['recipe[mdbci_provision_mark::remove_mark]',
+                "recipe[#{recipe_name}]",
+                'recipe[mdbci_provision_mark::default]']
+    if subscription_manager_params[:need_configure]
+      run_list.insert(1, subscription_manager_params[:recipe_name])
+      product_config = product_config.merge(subscription_manager_params[:credentials])
+    end
+    role = { name: name,
+             default_attributes: {},
+             override_attributes: product_config,
+             json_class: 'Chef::Role',
+             description: '',
+             chef_type: 'role',
+             run_list: run_list }
     JSON.pretty_generate(role)
   end
 
+  # Generate the subscription manager params for generating role JSON-file.
+  #
+  # @param box_params [Hash] information of the box parameters
+  # @param credentials [Hash] information of the box credentials
+  # @return [Hash] subscription manager params in format { need_configure, recipe_name, credentials }
+  # for generating role JSON-file.
+  def make_subscription_manager_params(box_params, credentials)
+    return { need_configure: false } unless box_params['configure_subscription_manager'] == 'true'
+
+    if credentials.nil?
+      @ui.error('RHEL credentials for Red Hat Subscription-Manager are not configured')
+      return { need_configure: false }
+    end
+
+    { need_configure: true,
+      recipe_name: 'recipe[subscription-manager]',
+      credentials: credentials }
+  end
+
   # Generate the role description for the specified node.
   #
   # @param name [String] internal name of the machine specified in the template
   # @param product [Hash] parameters of the product to configure from configuration file
-  # @param box information about the box
+  # @param box [String] name of the box
+  # @param box_params [Hash] information of the box parameters
```

Comment:
Why this method can not access the `@boxes` instance variable as other methods you have changed in this change set?